File: is/betbook/tags/it-1/Model/EventService/EventService.cs
using System;
using System.Collections.Generic;
using Microsoft.Practices.Unity;
using Es.Udc.DotNet.ModelUtil.Exceptions;
using Es.Udc.DotNet.ModelUtil.Transactions;
using Es.Udc.DotNet.Betbook.Model.BetOMaticService;
using Es.Udc.DotNet.Betbook.Model.FavouriteDao;
using Es.Udc.DotNet.Betbook.Model.RecommendationDao;
using Es.Udc.DotNet.Betbook.Model.UserDao;
using Es.Udc.DotNet.Betbook.Model.GroupDao;

namespace Es.Udc.DotNet.Betbook.Model.EventService
{
    class EventService: IEventService
    {
        [Dependency]
        public IBetOMaticService BetOMaticService { private get; set; }
        [Dependency]
        public IFavouriteDao FavouriteDao { private get; set; }
        [Dependency]
        public IRecommendationDao RecommendationDao { private get; set; }
        [Dependency]
        public IUserDao UserDao { private get; set; }
        [Dependency]
        public IGroupDao GroupDao { private get; set; }


        #region IEventService Members

        public List<Event> FindEvents(List<string> keywords, int startIndex, int count)
        {
            return BetOMaticService.FindEvents(keywords, startIndex, count);
        }

        /// <exception cref="InstanceNotFoundException"/>
        [Transactional]
        public void RecommendEvent(long eventId, List<long> groupIds, long userId, global::System.DateTime date)
        {
            Recommendation recommendation = Recommendation.CreateRecommendation(0, eventId, date);
            recommendation.eventId = eventId;
            recommendation.User = UserDao.Find(userId);
            foreach (long groupId in groupIds)
            {
                recommendation.Group.Add(GroupDao.Find(groupId));
            }

            RecommendationDao.Create(recommendation);
        }

        /// <exception cref="InstanceNotFoundException"/>
        public List<Recommendation> FindRecommendations(long userId)
        {
            return RecommendationDao.FindByUserId(userId);
        }


        public List<BetType> FindBetTypes()
        {
            return BetOMaticService.FindBetTypes();
        }

        /// <exception cref="InstanceNotFoundException"/>
        [Transactional]
        public void FavouriteEvent(long eventId, long userId, string name, string comment, global::System.DateTime date)
        {
            Favourite favourite = Favourite.CreateFavourite(0, date, eventId);
            favourite.eventId = eventId;
            favourite.comment = comment;
            favourite.name = name;
            favourite.User = UserDao.Find(userId);

            FavouriteDao.Create(favourite);
        }

        public List<Favourite> FindUserFavourites(long userId)
        {
            return FavouriteDao.FindByUserId(userId);
        }

        /// <exception cref="InstanceNotFoundException"/>
        [Transactional]
        public void RemoveFavourite(long favId)
        {
            FavouriteDao.Remove(favId);
        }

        #endregion
    }
}

File: is/bet-o-matic/tags/it-1/src/main/java/es/udc/betomatic/model/eventservice/EventService.java
package es.udc.betomatic.model.eventservice;

import java.util.Calendar;
import java.util.List;
import es.udc.pojo.modelutil.exceptions.InstanceNotFoundException;
import es.udc.betomatic.model.bettype.BetType;
import es.udc.betomatic.model.event.Event;

public interface EventService {

	public Event createEvent(Event event);
	public void addBetTypeToEvent(Long eventId, BetType betType)
			throws InstanceNotFoundException;
	public EventBlock findEvents(List<String> substrings,
			Calendar startDate, int startIndex, int count )
			throws InstanceNotFoundException;
	public EventBlock findEvents(List<String> substrings, long catId,
			Calendar startDate, int startIndex, int count )
			throws InstanceNotFoundException;
	public EventBlock findEventsAdmin(List<String> substrings,
			int startIndex, int count )
			throws InstanceNotFoundException;
	public EventBlock findEventsAdmin(List<String> substrings, long catId,
			int startIndex, int count )
			throws InstanceNotFoundException;
	public int getNumberOfEvents(List<String> substrings, Calendar startDate)
			throws InstanceNotFoundException;
	public int getNumberOfEvents(List<String> substrings, long catId, Calendar startDate)
			throws InstanceNotFoundException;
	public int getNumberOfEventsAdmin(List<String> substrings)
			throws InstanceNotFoundException;
	public int getNumberOfEventsAdmin(List<String> substrings, long catId)
			throws InstanceNotFoundException;

}
